s3-shield/cloud-init/generate_yaml.sh

```diff
@@ -6,8 +6,16 @@ cd $(dirname "$0")
 BASE64_VCL=$(base64 -w 0 "../default.vcl")
 BASE64_S3CONF=$(base64 -w 0 "s3.conf")
 
+if [ -n "$1" ]
+then
+  INSTALL_VARNISH_COMMAND="curl https:\/\/docs.varnish-software.com\/scripts\/setup\.sh \| TOKEN=$1 INSTALL=\"varnish\-plus\" bash"
+else
+  INSTALL_VARNISH_COMMAND="echo 'Not installing Varnish Enterprise, assuming it is already installed'"
+fi
+
```

Comment:
save yourself some quoting woes and transfer that logic to `runcmd`, and only `sed` the token value

Reply:
I don't think we need to run the setup script on a system that already has Varnish installed. Seems like a waste of time.

Reply:
I'm just saying to have the same if logic in `runcmd` so the `sed` doesn't have to deal with a complex string:

```
runcmd:
  - |
    TOKEN=INSTALL_VARNISH_TOKEN
    if [ -n "$TOKEN" ]; then
      curl ....
    else
      echo
    fi
```

Reply:
@gquintard The solution you provided indeed made things easier.